### ISSUE TITLE
Delete objects from cache if cached function not called

### ISSUE THREAD

**Issue by hardbyte** [JuliaServices/ExpiringCaches.jl#1]
I noticed that items only get deleted when the cache is queried, this might be a design decision but might be worth documenting. I was considering using it to store large amounts of data in memory that should be removed when the expiry is up but this package will currently only expire items when the key is queried.

Example:

```julia
cache = ExpiringCaches.Cache{Int64, String}(Dates.Second(5))
get(cache, 1, "hello")
@test length(cache) == 0
get!(cache, 1, "hello")
@test length(cache) == 1
@test get(cache, 1, "not found") == "hello"

# Wait for cache to expire
sleep(5)
@test length(cache) == 0
# Fails unless get(cache, 1, "") has been called before the length check.
```

If we call `get(cache, SOMEOTHERKEY, "")` that doesn't check for any other expiring keys. 

As far as I can see a user would have to iterate over all the keys and call `haskey` to trigger a cleanup.

**Comment by quinnj** [comment]
I can think of two options here (sorry for the slow response BTW): 1) provide a `ExpiringCaches.clear!(x)` function that would basically loop thru and purge expired keys or 2) provide a keyword argument that would start an async timer task for each key-val set that would remove the key when expired.

Any thoughts/preference?

**Comment by hardbyte** [comment]
Both approaches seem reasonable to me. I think 2 could be a better API and would be what I'd expect ExpiringCaches to behave like. 

Is there any need for the _auto purge_ argument to be set on a per key basis? I'd have thought it should be a once off Cache level configurable/type.

**Comment by quinnj** [comment]
Implemented in #7
